models/Asset.php

```diff
@@ -408,19 +417,10 @@ public static function getTypeFromMimeMapping(string $mimeType, string $filename
         }
 
         $type = null;
+        $assetTypes = Pimcore::getContainer()->getParameter('pimcore.config')['assets']['type_definitions']['map'];
 
-        $mappings = [
-            'unknown' => ["/\.stp$/"],
-            'image' => ['/image/', "/\.eps$/", "/\.ai$/", "/\.svgz$/", "/\.pcx$/", "/\.iff$/", "/\.pct$/", "/\.wmf$/", '/photoshop/'],
-            'text' => ['/text\//', '/xml$/', '/\.json$/'],
-            'audio' => ['/audio/'],
-            'video' => ['/video/'],
-            'document' => ['/msword/', '/pdf/', '/powerpoint/', '/office/', '/excel/', '/opendocument/'],
-            'archive' => ['/zip/', '/tar/'],
-        ];
-
-        foreach ($mappings as $assetType => $patterns) {
-            foreach ($patterns as $pattern) {
+        foreach ($assetTypes as $assetType => $assetTypeConfiguration) {
+            foreach ($assetTypeConfiguration['mapping'] as $pattern) {
```

Comment:
How to deal with types that are using the PrefixLoader, they are not able to provide a mapping then. Maybe we even don't need the PrefixLoader? 🤔